elytron/ELY-201-Add-client-side-HTTP-authentication-mechanism-support.adoc

```diff
@@ -0,0 +1,171 @@
+= ELY-201 Add client side HTTP authentication mechanism support. 
+:author:            Keshav Kumar
+:email:             your.email@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Wildfly Elytron already provide client and server side SASL mechanism for server side HTTP authentication mechanism.This will include a client side framework for HTTP authentication mechanisms and implement a set of client side HTTP authentication mechanisms.
```

Comment:
Is there some thought of using https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/Authenticator.html?

Reply:
@jamezp we did not discuss the Authenticator usage. But it seems that for our use case where we need to obtain not just a password from the elytron configuration but also the ssl context, credentials stored in the credential store, and bearer tokens it is not so clear how the Authenticator would fit in. Let us know if you have any idea or a suggestion for it

Reply:
Hmm... ...yeah I see what you mean. It's a bummer there isn't a better API for https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.Builder.html#authenticator(java.net.Authenticator).